### DISCUSSION TITLE
Allow Assets Field and Entries Field to be managed at the site, site group, and language level

### DISCUSSION THREAD

**Discussion by codyjames** [craftcms/cms#8539] [Ideas] [answered]
It would cool if Assets Fields (and Entries Fields) could be managed at the site, site group, and language level _in the same way as other fields_. For instance, we have an "English" site group and would want to use the same assets for all of those sites, but a French site may use a different asset.

<img width="333" alt="screen shot 2018-04-25 at 2 13 55 pm" src="https://user-images.githubusercontent.com/378665/39273482-c2876d5a-4893-11e8-86d4-aacf67d6eb59.png">

**Comment by codyjames** [comment]
We could really use this right now. We're in the process of translating a site into 12+ languages and the current options for assets really force us down some wonky/hacky paths.

**Comment by brandonkelly** [comment]
I’m not even sure how we’d go about this, as there is currently a direct correlation between assets and the physical files on the server.

One option is you could create a volume specifically for holding “translated” assets, and put a “Translated File” custom Assets field on your main asset volume, connected to the translated asset volume & set to manage relations on a per-site basis. Then use your main volume to hold the “primary” language’s assets, and whenever you need to supply a translated version of an asset, just double-click on it, and  upload a file to the Translated File field.

Then from your template, do this whenever dealing with an asset:

```twig
{% set asset = asset.translatedFile.one() ?? asset %}
```

Or if looping through multiple assets, you could eager load them.

```twig
{% set assets = entry.myAssetsField.with(['translatedFile']).all() %}
{% for asset in assets %}
    {% set asset = asset.myAssetsField[0] ?? asset %}
    ...
{% endfor %}
```

**Comment by codyjames** [comment]
Oops. I think I wasn't quite clear here, I'm talking about the Assets Field.

I want to be able to set an Assets Field on an Entry and have it be set for all sites in that Site Group (or whatever translation method I choose). Same goes for Entries Field.

Ideally all fields should have the same translation method options available.

**Comment by brandonkelly** [comment]
Ah, right, sorry I must have been thinking of a different issue.

Making relations translatable with the same modes as normal fields would be quite a big change, so not something we would do unless this FR got significant traction.

**Comment by jsunsawyer** [comment]
Without this feature, translations are really confusing to explain to the client. It immediately introduces an exception to how groups work.

User wants to edit an image and text field for all sites in an English site group.
- User updates text field content: Text is updated for each site in group.
- User updates an image: Image is updated for the current site, other sites in the group use the old image.

Outside of creating extra work, this inconsistency introduces a scenario where content can easily and unintentionally get out of sync.

This seems like a really important change to make translations clear and consistent.

**Comment by mats852** [comment]
This is a feature I stand behind for sure !

**Comment by michaelhue** [comment]
@brandonkelly Would you accept a pull request for this? I'm not very familiar with the internals of the relation fields but I'd be willing to give it a shot. A little guidance would go a long way though.

Update `3.2` added propagation methods for sections and matrix fields, which is fantastic for "true" multi-site + multi-language setups. The two relation fields being an exception hurts the experience.

I completely agree with @jsunsawyer; clients are repeatedly confused by this.

**Comment by brandonkelly** [comment]
@michaelhue I think it’s best if this one is resolved in-house. We are going to be making other content modeling-related changes in Craft 4, so I think that would be the best time to try to tackle this. I’ll add it to that milestone and see what I can come up with.

**Comment by timkelty** [comment]
To clarify: you already can manage these per-site, right? ("Manage relations on a per-site basis" checkbox in field settings)

So the issue remains for how to manage them on the group level, right?

I use https://github.com/timkelty/craftcms-site-sync to get around some of these issues. Not sure if it will fit your use-case, though.

**Comment by wsydney76** [comment]
>  so not something we would do unless this FR got significant traction.

Happy to add some traction. A lot. The lack of this is leading to a lot of confusion/inconsistency/epic instructions/strange workarounds.

**Comment by soschubert** [comment]
Looks like this is still unsolved?

We want to offer some files for download and they are in different languages. Almost everything else is 'translatable' but not the asset content/field.

We used a workaround for now, but we'd be happy if there is a clean solution for it in the future.

**Comment by andris-sevcenko** [comment]
@soschubert there will be, but for it to be clean, there needs to be a refactor of how content modeling works in Craft. Currently looks to be slated for 5.0.

**Comment by johnwbaxter** [comment]
I found this one a bit confusing when creating a multi site site. Coming from EE land, I was expecting everything to be interchangeable so this threw me a bit :)

To get around this, I wrap fields like this in a supertable field so that it's translatable, but I do think that they should be translatable like everything else.

**Comment by maxstrebel** [comment]
Thats a clever hack @johnwbaxter. 

I sincerely hope you consider a Refactor in 4.0, @andris-sevcenko. After my last Multisite Project, I'm mildly frustrated (because it is a bump in an otherwise very positive client experience). Translations and also Asset Paths on a Site basis are all big hacks right now.

**Comment by jalendport** [comment]
Gonna add a +1 to this; just ran into a usecase where the ability to translate asset fields per group would be super helpful...

Thanks for the workaround @johnwbaxter; gonna have to roll with that for now 🙌

**Comment by jacksoncool** [comment]
bump... This would be really useful especially for globals.

**Comment by cmalven** [comment]
Running into this on a current project. Each site should have it's own "logo" asset, but there is no easy way to manage this with a single asset field that is shared between sites.

**Reply by mats852** [reply]
In the meantime, for static assets, I do something like that:

```twig
<img src="/static/svg/logo_{{ craft.app.language |split('-') |first }}.svg" alt="" />
```

**Reply by jalendport** [reply]
@cmalven I _think_ what you're describing is a little different from what this issue is about. I'm fairly certain you can achieve what you're describing with one asset field set to "Manage relations on a per-site basis" that you stick in a global perhaps. What we're asking for in this issue are options to "Manage relations on a per-site-group basis" and "Manage relations on a per-site-language basis".

**Reply by cmalven** [reply]
@jalendport That may be. It seemed from the initial message that kicked this off and some of the comments, that the request was that `Asset` fields had the same translation options as other fields. e.g. `Not Translatable`, `Translate for Each Site Group`, etc. I may be able to achieve what I'm trying to using `Manage relations on a per-site basis`. I didn't realize that existed until you mentioned it just now.

**Reply by brandonkelly** [reply]
@cmalven Yep that’s exactly what you need.

**Reply by animaux** [reply]
Yey! `Manage relations on a per-site basis`, thanks! `\o/`

**Comment by dr-codswallop** [comment]
Another +1 request for this. 

I'm also using the workaround from @johnwbaxter for now, but would be really helpful for asset fields to have the same translation options as other fields natively. Especially in my current use case for choosing an image per **site group** in a global.

**Comment by oriolnoya** [comment]
Another +1 request for this.

Meanwhile, I'm using the following workaround: I wrap the asset in a "Link field" from this plugin: https://github.com/sebastian-lenz/craft-linkfield

**Comment by Antimated** [comment]
+1 on this. I need to add a logo for each site group. Each site group has 5 languages. So if the logo changes the client has to change the logo 5 times :/

**Reply by brandonkelly** [reply]
Alternatively, you could create separate Assets fields for each site group, and then set each of the fields to only show for their intended site group, via field conditions in the field layout designer.

**Comment by mark-wff** [comment]
+1 on this. 

This is a requirement that we have in almost all of our projects. To achieve translation in different groups or languages, we use a Matrix-Field with a File-Field. It would be great if asset-fields could be translated as a native feature

**Comment by brandonkelly** [comment, marked as answer]
Relation fields are gaining full Translation Method support in Craft 5.3 (#15400) 🎉

**Reply by brandonkelly** [reply]
Craft 5.3.0-beta.1 is out with that change 🎉 (#15400)

We’re targeting August 6 for the GA release. If you’d like to test ahead of time, change your `craftcms/cms` requirement in composer.json to:

```json
"craftcms/cms": "^5.3.0-beta.1",
```

and then run `composer update`.
